### PR TITLE
test: add e2e tests for `client` option

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Yes

### Motivation / Use-Case

add e2e tests for `client` option.

### Breaking Changes

None

### Additional Info
No

### CODE REVIEW

File: test/e2e/client.test.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+      {
+        title: 'as a path ("sockjs")',
+        client: {
+          webSocketTransport: require.resolve(
+            "../../client-src/clients/SockJSClient"
+          ),
+        },
+        webSocketServer: "sockjs",
+        shouldThrow: false,
+      },
```
Comment (alexander-akait) [inline test/e2e/client.test.js:216]: Let's add test for ws here too

Comment (snitin315) [reply test/e2e/client.test.js:224]: Done.

File: test/e2e/client.test.js [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        client: {
+          webSocketTransport: require.resolve(
+            "../../client-src/clients/WebSocketClient"
+          ),
+        },
+        webSocketServer: "ws",
+        shouldThrow: false,
+      },
+      {
+        title: "as a nonexistent path",
```
Comment (snitin315) [inline test/e2e/client.test.js:236]:
```suggestion
        title: "as a nonexistent path (sockjs)",
```

File: test/e2e/client.test.js [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        client: {
+          webSocketTransport: "/bad/path/to/implementation",
+        },
+        webSocketServer: "sockjs",
+        shouldThrow: true,
+      },
+      {
+        title: "as a nonexistent path",
```
Comment (snitin315) [inline test/e2e/client.test.js:244]:
```suggestion
        title: "as a nonexistent path (ws)",
```

**Pull request merged**